cordyceps/src/lib.rs

```diff
@@ -59,6 +62,15 @@
 //!   A [`TransferStack`] can be used to efficiently transfer ownership of
 //!   resources from multiple producers to a consumer, such as for reuse or
 //!   cleanup.
+//!
+//!   This structure is only available if the target supports CAS (Compare and
+//!   Swap) atomics.
+//!
+//! ## Compatibility
+//!
+//! Rudimentary support for targets without CAS (Compare and Swap) atomics, such as
+//! Cortex-M0+/`thumbv6m-none-eabi`, is provided, however not all structures and
+//! features may be available.
```

Comment:
Nit, take it or leave it: it would maybe be nice if this explicitly stated that this is detected via `target_has_atomic = "ptr"`, with a reference to the docs for that? It seems like it could be helpful for readers to know exactly what cfg attribute detects this.